Source/SwiftLintFramework/Rules/UnusedOptionalBindingRule.swift

```diff
@@ -40,33 +40,29 @@ public struct UnusedOptionalBindingRule: ASTRule, ConfigurationProviderRule {
     public func validateFile(_ file: File,
                              kind: StatementKind,
                              dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
-        guard kind == .if || kind == .guard else {
-                return []
+        guard kind == .if || kind == .guard,
+            let offset = (dictionary["key.offset"] as? Int64).flatMap({ Int($0) }),
+            let length = (dictionary["key.length"] as? Int64).flatMap({ Int($0) }),
+            let range = file.contents.bridge().byteRangeToNSRange(start: offset, length: length) else {
+            return []
         }
 
-        return violationRanges(file: file).map {
+        return violationRanges(file: file, in: range).map {
             return StyleViolation(ruleDescription: type(of: self).description,
                            severity: configuration.severity,
                            location: Location(file: file, characterOffset: $0.location))
         }
     }
 
-    private func violationRanges(file: File) -> [NSRange] {
+    private func violationRanges(file: File, in range: NSRange) -> [NSRange] {
         let kinds = SyntaxKind.commentAndStringKinds()
         let underscorePattern = "\\b_\\b"
         let parenthesesPattern = "\\([^)]*\\)"
-        let pattern = underscorePattern + "|" + parenthesesPattern
+        let pattern = underscorePattern
```

Comment:
this assignment is not needed, you can just use `underscorePattern` in the call below